### PR TITLE
build/pkgs/tox: Update, require tox >= 4.2.7, update dependencies

### LINKED ISSUE #34840: Upgrade tox to 4.x
<div id="comment:0"></div>

Depends on #34839

Component: **packages: standard**

Branch: **[u/mkoeppe/upgrade_tox_4](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/upgrade_tox_4)**

_Issue created by migration from https://trac.sagemath.org/ticket/34840_

### LINKED ISSUE #36175: toml package is obsolete
### Steps To Reproduce

_No response_

### Expected Behavior

toml is not installed

### Actual Behavior

toml is installed

### Additional Information

I think our `toml` package is obsolete. The only other package listing it as a requirement is tox, but tox only needs it with ancient pythons: https://github.com/tox-dev/tox/blob/3.27.1/setup.cfg#L45

I think our dependency can be replaced with tomli at which point toml can go.

### Environment

```markdown
Gentoo / git develop
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by mkoeppe**
- Fixes #34840
- Fixes #36175

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

- Depends on #36240 (merged here)
- Depends on #36263 (merged here)

### CODE REVIEW

File: build/pkgs/virtualenv/SPKG.rst [hunk 1]
```diff
@@ -9,8 +9,6 @@ Virtual Python Environment builder
 License
 -------
 
-MIT
-
 Upstream Contact
```
Comment (kwankyu) [inline build/pkgs/virtualenv/SPKG.rst]: Why is the license info deleted?

File: build/pkgs/tox/SPKG.rst [hunk 2]
```diff
@@ -9,8 +9,6 @@ tox is a generic virtualenv management and test command line tool
 License
 -------
 
-MIT
-
 Upstream Contact
```
Comment (kwankyu) [inline build/pkgs/tox/SPKG.rst]: Why the license info is deleted? It seems still valid.

**Comment by kwankyu** [comment]
It looks good to me as far as I check.

I will set positive review after Build & Test passes here and locally, unless @orlitzky chimes in.

**Review by kwankyu — APPROVED** [review]
It works well with me.

**Comment by vbraun** [comment]
merge conflict

**Comment by kwankyu** [comment]
Is "Build&Test using Conda" supposed to pass? What should I compare it with? Does it pass on develop branch?

**Comment by mkoeppe** [comment]
"Build&Test using Conda" has been broken for a while now; a possible fix in #35986 is stuck

**Comment by kwankyu** [comment]
OK then.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**